lightning/src/ln/chan_utils.rs

```diff
@@ -550,11 +556,430 @@ pub fn build_htlc_transaction(prev_hash: &Txid, feerate_per_kw: u32, contest_del
 	}
 }
 
+// Per-channel data used to build transactions given per-commitment data (CommitmentTransaction)
+#[derive(Clone)]
+pub(crate) struct ChannelTransactionParameters {
+	// Holder public keys
+	pub(crate) holder_pubkeys: ChannelPublicKeys,
+	// Counter-party public keys
+	pub(crate) counterparty_pubkeys: ChannelPublicKeys,
+	// The contest delay selected by the holder
+	pub(crate) holder_selected_contest_delay: u16,
+	// The contest delay selected by the counterparty
+	pub(crate) counterparty_selected_contest_delay: u16,
+	// The funding outpoint
+	pub(crate) funding_outpoint: OutPoint,
+	// Whether the channel is outbound from the point of view of the holder
+	pub(crate) is_outbound_from_holder: bool,
+}
+
+impl ChannelTransactionParameters {
+	pub(crate) fn to_directed(&self, holder: bool) -> DirectedChannelTransactionParameters {
+		DirectedChannelTransactionParameters {
+			info: self,
+			broadcaster_is_holder: holder
+		}
+	}
+}
+
+impl_writeable!(ChannelTransactionParameters, 0, {
+	holder_pubkeys,
+	counterparty_pubkeys,
+	holder_selected_contest_delay,
+	counterparty_selected_contest_delay,
+	funding_outpoint,
+	is_outbound_from_holder
+});
+
+/// Static channel fields used to build transactions given per-commitment fields (CommitmentTransaction)
+pub struct DirectedChannelTransactionParameters<'a> {
```

Comment:
Thanks for this new structure, you can add a comment explaining the order : "If holder is the broadcaster, we map structure elements (holder_pubkeys, counterparty_pubkeys, counterparty_selected_contest_delay) to (broadcaster_pubkeys, countersignatory_pubkeys, countersignatory_contest_delay). This structure make transaction fields as consumed by CommittmentTransaction builders independent of transaction ownership.

Reply:
done, and added a few more words about the other direction